.github/workflows/private-package.yml

```diff
@@ -0,0 +1,47 @@
+name: provider-upgrade
+
+on:
+  schedule:
+    - cron: '0 12 * * *'
+  pull_request: {}
+  workflow_dispatch: {}
+
+env:
+  GO_VERSION: '1.18'
+  KIND_VERSION: 'v0.14.0'
+
+jobs:
+  provider-upgrade-crossplane-stable:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Test Repo
+        uses: actions/checkout@v2
+      - name: Setup Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: ${{ env.GO_VERSION }}
+      - name: Setup Kind
+        uses: engineerd/setup-kind@v0.5.0
+        with:
+          version: ${{ env.KIND_VERSION }}
+      - name: Install Helm
+        uses: azure/setup-helm@v1
+      - name: Install Kubectl
+        uses: azure/setup-kubectl@v1
+      - name: Create Namespace
+        run: kubectl create namespace crossplane-system
+        shell: bash
+      # https://cloud.google.com/artifact-registry/docs/access-control#pullsecrets
+      - name: Create Namespace
+        run: |
+          kubectl create secret docker-registry package-pull-secret -n crossplane-system \
+          --docker-server=us-west1-docker.pkg.dev \
+          --docker-email=${{ secrets.GAR_EMAIL }} \
```

Comment:
@jbw976 can you add the `GAR_EMAIL` secret to this repo, which should match the email in the `GAR_SA_KEY`?

Reply:
yes sir, i will do that right now!!